packages/wallets/nightly/src/adapter.ts

```diff
@@ -0,0 +1,174 @@
+
+    private _connecting: boolean;
+    private _connected: boolean;
+    private _publicKey: PublicKey | null;
+    private _wallet: SolanaNightly | null;
+    private _readyState: WalletReadyState =
+        typeof window === 'undefined' || typeof document === 'undefined'
+            ? WalletReadyState.Unsupported
+            : WalletReadyState.NotDetected;
+
+    constructor() {
+        super();
+        this._connecting = false;
+        this._connected = false;
+        this._publicKey = null;
+        this._wallet = null;
+
+        if (this._readyState !== WalletReadyState.Unsupported) {
+            scopePollingDetectionStrategy(() => {
+                if (window?.nightly?.solana) {
+                    this._readyState = WalletReadyState.Installed;
+                    this.emit('readyStateChange', this._readyState);
+                    return true;
+                }
+                return false;
+            });
+        }
+    }
+
+    get connected(): boolean {
+        return this._connected;
+    }
+
+    get connecting(): boolean {
+        return this._connecting;
+    }
+
+    get readyState(): WalletReadyState {
+        return this._readyState;
+    }
+
+    get publicKey() {
+        return this._publicKey;
+    }
+
+    async connect() {
+        try {
+            if (this.connected || this.connecting) return;
+            if (this._readyState !== WalletReadyState.Installed) throw new WalletNotReadyError();
+
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const wallet = window!.nightly!.solana!;
+
+            try {
+                this._connecting = true;
+                await wallet.connect();
+            } catch (error: any) {
+                if (error instanceof WalletError) throw error;
```

Comment:
`error instanceof WalletError` will never be true (the error is coming from the wallet), so this check isn't needed.

Reply:
removed